source/server/filter_chain_manager_impl.h

```diff
@@ -71,8 +77,37 @@ class FilterChainFactoryContextImpl : public Configuration::FilterChainFactoryCo
   Configuration::TransportSocketFactoryContext& getTransportSocketFactoryContext() const override;
   Stats::Scope& listenerScope() override;
 
+  void setDraining() { is_draining_.store(true, std::memory_order_release); }
+
 private:
   Configuration::FactoryContext& parent_context_;
+  Init::Manager& init_manager_;
+  // atomic_flag is guaranteed to be lock free but there is no light weight load().
```

Comment:
nit: del comment (we don't need to document the atomic behavior, we don't do this anywhere else).